torch/_functorch/partitioners.py

```diff
@@ -1734,6 +1736,9 @@ def classify_nodes(joint_module):
         for node in joint_module.graph.nodes:
             if node.op == "placeholder" and "tangents" in node.target:
                 required_bw_nodes.add(node)
+            elif _has_tag_must_be_in_backward(node):
+                required_bw_nodes.add(node)
```

Comment:
Is this fixing a bug in the partitioner that is technically independent of effect tokens? (it would be nice if we could add a test in a fresh PR that involves mutations in the backward in some way that is fixed by this, independently of the changes for getting effect-tokens-in-backward to work). If this is a pain though then np.

Reply:
Yes, will be good to have a separate PR with test cases, but can be difficult to make the needed graph with mutations in backward.

Reply:
Realized that the bug happens, when is_must_be_in_backward node is picked as a saved_value by partitioner. We do not have control to specify to partitioner which nodes to pick, so the test will be flaky. Once we have effectful ops as "must_be_in_backward" - it will be easier to make a test (like added here with_effects_only_in_backward)